### ISSUE TITLE
False positive with `Final` and read-only property in protocol

### ISSUE THREAD

**Issue by yaxum62** [microsoft/pyright#8744]
**Describe the bug**
Consider following code:
```python
from dataclasses import dataclass, field
from typing import Final, Protocol


@dataclass
class A:
    value: Final[int] = field(default=42, init=False)


class B:
    value: Final[int] = 42


class C:
    value: Final[int]

    def __init__(self):
        self.value = 42

class D:
    value: int = 42


class P(Protocol):
    @property
    def value(self) -> int: ...

def take_p(p: P):
    pass

take_p(A())
take_p(B())
take_p(C())
take_p(D())
```

Current pyright version reports errors on first three invocations of `take_p`:
```
Argument of type "A" cannot be assigned to parameter "p" of type "P" in function "take_p"
  "A" is incompatible with protocol "P"
    "value" is not marked Final in protocol
    "value" is not defined as a ClassVar in protocol

Argument of type "B" cannot be assigned to parameter "p" of type "P" in function "take_p"
  "B" is incompatible with protocol "P"
    "value" is not marked Final in protocol
    "value" is not defined as a ClassVar in protocol

Argument of type "C" cannot be assigned to parameter "p" of type "P" in function "take_p"
  "C" is incompatible with protocol "P"
    "value" is not marked Final in protocol
```
**Expected behavior**
Only the second error is valid, the first and third error seems like false positives.

The value field in dataclass `A` is assigned with field descriptor. Semantically it should be more close to class `C` instead of class `B`, thus `Final` should not imply `ClassVar` in this case.

Also `Final` fields without `ClassVar` should be compatible with read-only properties in protocol, with same reason as described in #2371 

**VS Code extension or command-line**
vs code extension PyLance v2024.8.1

**Comment by erictraut** [comment]
Thanks for the bug report. I agree with your analysis. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.376.
